persistent-postgresql/Database/Persist/Postgresql/Internal.hs

```diff
@@ -281,4 +285,23 @@ instance PersistField PgInterval where
 instance PersistFieldSql PgInterval where
   sqlType _ = SqlOther "interval"
 
-
+-- | Create an index for the given 'EntityField'.
+--
+-- This function returns a 'Migration', and so can be sequenced with another migration in a `do` block.
+--
+-- @
+-- migrations :: 'Migration'
+-- migrations = do
+--     migrateAll
+--     createSearchIndex UserName
+-- @
+--
+-- @since 2.13.3.0
+createSearchIndexPostgres :: forall rec typ. (PersistEntity rec) => EntityField rec typ -> Migration
+createSearchIndexPostgres entityField =
+  addMigration True $ "CREATE INDEX IF NOT EXISTS "
```

Comment:
I'm hesitant to have an API that will break if you use the default function - in these cases, I either prefer to only have the `persistent-{backend}` libraries exporting these specialized functions, or make this a field on `SqlBackend` to allow each backend to do this properly.